packages/stream-tests/src/__tests__/basic-indexing.test.ts

```diff
@@ -521,6 +519,224 @@ describe.each(envs)('Basic end-to-end indexing query test for $dbEngine', (env)
     })
   })
 
+  describe('queries with custom sorting', () => {
+    test('multiple documents - one page - ASC order with forward pagination', async () => {
+      const doc0 = await ModelInstanceDocument.create(ceramic, CONTENT0, midMetadata)
+      const doc1 = await ModelInstanceDocument.create(ceramic, CONTENT1, midMetadata)
+      const doc2 = await ModelInstanceDocument.create(ceramic, CONTENT2, midMetadata)
+
+      const resultObj = await ceramic.index.query({
+        model: model.id,
+        sorting: { myData: 'ASC' },
```

Comment:
I want to make sure we leave room for this to be extensible.  Like you might want to be able to specify more complex sorting behavior than just a set of content fields.  For example, you might want to sort on anchored timestamp or updated_at timestamp instead of created_at.  You also might want a combination of sorting on custom content fields PLUS some control over built-in fields like created_at, updated_at, etc.

So as to leave this more extensible for the future, I don't think we should put the custom fields at the top level of the sorting object.  But instead putting it behind a `byFields` parameter so it looks like `sorting: { byFields: { myData: 'ASC' } }` would leave us the option of adding other reserved top-level arguments to the sorting object in the future that could further fine-tune the behavior.

Reply:
now that I think about it, the same argument could be made for the filters themselves.  Like, do we have a way today to filter on `created_at` greater than some timestamp?  If not, we should think about how we might add such capability in the future and make sure our syntax/api is extensible enough to allow that.

Reply:
I think we should leave the `created_at` and `updated_at` abstracted from the consumers. These values are defined in the DB, not across the network so the behavior could be confusing, I'd much rather make it explicit that these need to be defined in the contents if apps want to filter and/or order by them.

Reply:
Regarding future-proofing, what we've done so far is adding fields to the `BaseQuery` object, for example we have the `filter` object that we're using for relations support, and the new `queryFilters` object with custom filters, so we could have the `sorting` field as-is for now and add a new object if we need to change the shape?
Eventually it can be abstracted at the GraphQL level anyway so I don't mind having a `byFields` object in `sorting` if you prefer, the logic wouldn't change much either way.

Reply:
yeah, I kind of hate that we have both `filter` and `queryFilters`.  I really wonder if we can't combine them into one?
By the same token I slightly prefer pushing down a byFields subobj into `sorting` so that `sorting` itself can be extended, but you're right that a new top-level option could accomplish the same goal so I ultimately don't feel strongly either way